test/performance/driver.F90

```diff
@@ -0,0 +1,153 @@
+! Copyright (C) 2022 National Center for Atmospheric Research
+! SPDX-License-Identifier: Apache-2.0
+!
+!> \file
+!> A driver for performance evaluation of MICM
+program performance_test
+
+  use musica_constants,                only : dk => musica_dk
+
+  implicit none
+
+  integer, parameter :: kNumberOfGridCells = 100
+  integer, parameter :: kNUmberOfTimeSteps = 100
+  real(kind=dk), parameter :: kTimeStep__min = 5.0_dk
+
+  call run_test( )
+
+contains
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  !> Runs MICM under prescribed conditions and collect performance metrics
+  subroutine run_test( )
+
+    use micm_environment,              only : environment_t
+    use micm_kinetics,                 only : kinetics_t
+    use micm_ODE_solver,               only : ODE_solver_t
+    use micm_ODE_solver_factory,       only : ODE_solver_builder
+    use musica_assert,                 only : assert_msg
+    use musica_config,                 only : config_t
+    use musica_string,                 only : string_t, to_char
+
+    character(len=*), parameter :: my_name = "MICM performance test"
+    type(config_t) :: solver_config
+    class(ODE_solver_t), pointer :: solver => null( )
+    class(kinetics_t), pointer :: kinetics => null( )
+    type(string_t), allocatable :: species_names(:)
+    type(string_t), allocatable :: reaction_names(:)
+    type(string_t), allocatable :: photo_reaction_names(:)
+    ! Species number densities [molecule cm-3] (grid cell, species)
+    real(kind=dk), allocatable :: number_densities__molec_cm3(:,:)
+    type(environment_t) :: env(kNumberOfGridCells)
+    integer :: i_time, error_flag
+
+    ! Set up the kinetics calculator
+    kinetics => kinetics_t( )
+    call kinetics%species_names( species_names )
+    call kinetics%reaction_names( reaction_names )
+    call kinetics%photolysis_reaction_names( photo_reaction_names )
+
+    ! Set up the solver for the test
+    call solver_config%empty( )
+    call solver_config%add( "type", "Rosenbrock", my_name )
+    call solver_config%add( "chemistry time step", "min", kTimeStep__min,     &
+                            my_name )
+    call solver_config%add( "absolute tolerance", 1.0e-12, my_name )
+    call solver_config%add( "relative tolerance", 1.0e-4, my_name )
+    call solver_config%add( "number of variables", size( species_names ),     &
+                            my_name )
+    ! TODO determine if the solver needs to know the number of grid cells
+    !      during initialization
+    solver => ODE_solver_builder( solver_config )
```

Comment:
I would only send this info if it's needed, but you could just add an argument for the number of cells to the `solver_config` and modify the solver constructor. Ideally we wouldn't need to send this info, and I think the most of the preprocessed code could avoid asking for `ncells` and just use the dimensions of the passed arrays. But, this is just for a test, so I wouldn't spend too much time on a robust design

Reply:
Great point! I agree that we do not need to pass `ncell` as an argument at all. This could simplify our changes here. I will revise the preprocessed code later to reflect your suggestion here.

Reply:
Not passing the ncell value through the call stack will negatively impact the CPU's performance.  It will impact the ability of the compiler to generate efficient vectorized code.  I don't know how it will affect the GPU case.  We can always test and potentially modify in the final version of the solver.

Reply:
Thanks @johnmauff. I have heard the similar statement about the negative impact of assumed-shape array on the performance. I think it will be a good sensitive test in the future work but right now we need to get the static MICM work with the multi-gridpoint chemistry solver first.